### PR TITLE
recipe for pathological

### PR DESCRIPTION
**Pull request by Vrondir**

### CODE REVIEW

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 1]
```diff
@@ -0,0 +1,42 @@
+SUMMARY="An enriched clone of the game Logical."
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: No "dot" at the end

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+DESCRIPTION="Pathological is an enriched clone of the game \
+""Logical"" by Rainbow Arts. \
+To solve a level, fill each wheel with four marbles of matching color. \
+Various board elements such as teleporters, switches, filters, etc., \
+make the game interesting and challenging. \
+New levels can be created using your favorite text editor.."
+HOMEPAGE="http://pathological.sourceforge.net/"
+COPYRIGHT=""
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Put "Unknown" for now

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/pathological/files/pathological/1.1.3/pathological-1.1.3.tar.gz"
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Use $portVersion if you can:
```diff
-SOURCE_URI="https://sourceforge.net/projects/pathological/files/pathological/1.1.3/pathological-1.1.3.tar.gz"
+SOURCE_URI="https://sourceforge.net/projects/pathological/files/pathological/$portVersion/pathological-$portVersion.tar.gz"
```
Missing CHECKSUM_SHA256:
```diff
+CHECKSUM_SHA256="<checksum>"
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	pathological = $portVersion
+	"
+REQUIRES="
+		haiku
+		"
+
+BUILD_REQUIRES="
+		haiku_devel
+		"
+
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: No empty line between BUILD_REQUIRES and BUILD_PREREQUIRES
https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 5]
```diff
@@ -0,0 +1,41 @@
+SUMMARY="An enriched clone of the game Logical"
+DESCRIPTION="Pathological is an enriched clone of the game \
+"Logical" by Rainbow Arts. \
+To solve a level, fill each wheel with four marbles of matching color. \
+Various board elements such as teleporters, switches, filters, etc., \
+make the game interesting and challenging. \
+New levels can be created using your favorite text editor.clea"
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Suggestion:
```diff
-SUMMARY="An enriched clone of the game Logical"
-DESCRIPTION="Pathological is an enriched clone of the game \
-"Logical" by Rainbow Arts. \
-To solve a level, fill each wheel with four marbles of matching color. \
-Various board elements such as teleporters, switches, filters, etc., \
-make the game interesting and challenging. \
-New levels can be created using your favorite text editor.clea"
+SUMMARY="A puzzle game consisting of marbles that roll along paths"
+DESCRIPTION="Pathological is an enriched clone of the game Logical by Rainbow \
+Arts.
+To solve a level, fill each wheel with four marbles of matching color.
+Various board elements such as teleporters, switches, filters, etc., make the \
+game interesting and challenging.
+New levels can be created using your favorite text editor."
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        "
+
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+
+BUILD()
+{
+	make $jobArgs
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: You could use DESTDIR here to set a path to $appsDir/Pathological, that way it should be ok to build, with one addition ...

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+INSTALL()
+{
+	make install
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Same here for DESTDIR

File: games-puzzle/pathological/pathological-1.1.3.recipe (hunk 5 above)
Comment (Begasus) [reply games-puzzle/pathological/pathological-1.1.3.recipe]: Keeping the 80 characters restriction and create new lines (without "\" at the end)

File: games-puzzle/pathological/pathological-1.1.3.recipe (hunk 4 above)
Comment (Begasus) [reply games-puzzle/pathological/pathological-1.1.3.recipe]: Still an empty line here

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+COPYRIGHT="2003 John-Paul Gignac"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/pathological/files/pathological/$portVersion/pathological-$portVersion.tar.gz"
+CHECKSUM_SHA256="813ef0812ba6b3218b35c2ddea423c9c6ee912925824497f52d5310bc76f85de"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+
+PROVIDES="
+	pathological = $portVersion
```
Comment (korli) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: something should be provided here. see other recipes for examples.

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+game interesting and challenging. \
+New levels can be created using your favorite text editor."
+HOMEPAGE="http://pathological.sourceforge.net/"
+COPYRIGHT="2003 John-Paul Gignac"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/pathological/files/pathological/$portVersion/pathological-$portVersion.tar.gz"
+CHECKSUM_SHA256="813ef0812ba6b3218b35c2ddea423c9c6ee912925824497f52d5310bc76f85de"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Mark all architectures as "?" or even better as "!" (broken) for the PR

**Comment by Begasus** [comment]
Basic recipe is OK, to change the hard coded paths it will need a full re write of the makefile plus some sed magic to the python file

**Comment by korli** [comment]
> Basic recipe is OK, to change the hard coded paths it will need a full re write of the makefile plus some sed magic to the python file

Why WIP then?

**Comment by Begasus** [comment]
Well I could remove the WIP and merge, create an issue for the makefile, or leave it open?

**Comment by Begasus** [comment]
@korli I would be OK if this gets closed without merging, if someone picks it up later it maybe could be fixed, this shouldn't end up in the repository IMHO.

**Comment by pulkomandy** [comment]
Why would we close it and risk losing the work? Mark the recipe as broken, add a comment in it telling what still needs to be done, and merge it. Broken recipes in tree are ok, that's why we have these "broken" markers, after all?

**Comment by Begasus** [comment]
OK, recipe is already marked as broken, only needs a message in the recipe then to tell what is left to do.

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+
+BUILD()
+{
+	make DESTDIR=$appsDir/Pathological $jobArgs
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Maybe add a message here that the paths should be fixed in the makefile?
```
# paths are hardcoded in the makefile, they should point to the correct installation paths
```
(didn't check the 80 char limit here, if need be split up in 2 lines starting with "# ")

Comment (Vrondir) [reply games-puzzle/pathological/pathological-1.1.3.recipe]: Added the messages @Begasus :)

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 11]
```diff
@@ -0,0 +1,43 @@
+SUMMARY="A puzzle game consisting of marbles that roll along paths"
+DESCRIPTION="Pathological is an enriched clone of the game Logical by Rainbow \
+Arts. \
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Minux backslash, that way you new line starts on a new line :)
```diff
-Arts. \
+Arts.
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+To solve a level, fill each wheel with four marbles of matching color. \
+Various board elements such as teleporters, switches, filters, etc., make the \
+game interesting and challenging. \
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Same here:
```diff
-game interesting and challenging. \
+game interesting and challenging.
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 13]
```diff
@@ -0,0 +1,43 @@
+SUMMARY="A puzzle game consisting of marbles that roll along paths"
+DESCRIPTION="Pathological is an enriched clone of the game Logical by Rainbow \
+Arts. \
+To solve a level, fill each wheel with four marbles of matching color. \
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Same here:
```diff
-To solve a level, fill each wheel with four marbles of matching color. \
+To solve a level, fill each wheel with four marbles of matching color.
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+
+BUILD()
+{
+	#paths are hardcoded in the makefile,
+	#they should point to the correct installation paths
+	make DESTDIR=$appsDir/Pathological $jobArgs
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: As mentioned, DESTDIR not needed here:
```diff
-	make DESTDIR=$appsDir/Pathological $jobArgs
+	make $jobArgs
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+INSTALL()
+{
+	make DESTDIR=$appsDir/Pathological install
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: Usually we turn these around:
```diff
-	make DESTDIR=$appsDir/Pathological install
+	make install DESTDIR=$appsDir/Pathological
```

File: games-puzzle/pathological/pathological-1.1.3.recipe [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+game interesting and challenging.
+New levels can be created using your favorite text editor."
+HOMEPAGE="http://pathological.sourceforge.net/"
+COPYRIGHT="2003 John-Paul Gignac"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://sourceforge.net/projects/pathological/files/pathological/$portVersion/pathological-$portVersion.tar.gz"
+CHECKSUM_SHA256="813ef0812ba6b3218b35c2ddea423c9c6ee912925824497f52d5310bc76f85de"
+
+ARCHITECTURES="!x86_gcc2 !x86 x86_64"
```
Comment (Begasus) [inline games-puzzle/pathological/pathological-1.1.3.recipe]: You forgot to mark x86_64 as broken:
```diff
-ARCHITECTURES="!x86_gcc2 !x86 x86_64"
+ARCHITECTURES="!x86_gcc2 !x86 !x86_64"
```

**Pull request merged**